fpmsyncd/routesync.cpp

```diff
@@ -674,6 +674,21 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
     string nexthops = getNextHopGw(route_obj);
     string ifnames = getNextHopIf(route_obj);
 
+    vector<string> alsv = tokenize(ifnames, ',');
+    for (auto alias : alsv)
+    {
+        /*
+         * An FRR behavior change from 7.2 to 7.5 makes FRR update default route to eth0 in interface
+         * up/down events. Skipping routes to eth0 or docker0 to avoid such behavior
+         */
+        if (alias == "eth0" || alias == "docker0")
+        {
+            SWSS_LOG_NOTICE("Skip routes to eth0 or docker0: %s %s %s",
+                    destipprefix, nexthops.c_str(), ifnames.c_str());
+            return;
```

Comment:
Take this as a example: At first route 0.0.0.0/0  had a multipath nexthop "Ethernet0,Ethernet1", and some minutes later it became "Ethernet0,Ethernet1,eth0". Last it becomes "eth0". Since the change filters all routes including the eth0 nexthop, in the example this changes prevent "Ethernet0,Ethernet1" from exiting the next-hop list in ASIC. This is a rare situation but I suggest to take this into consideration.

Reply:
in our design, we would not never has such mixed scenario.